### PR TITLE
P7: Editors & Complex — residual audit cleanup (chat / markdown / review / JSE)

### PR DESCRIPTION
**Pull request by stevekinney**
## Summary

Residual accessibility/UX audit cleanup across the editor & complex components, in five independently-verified slices:

1. **Chat action buttons** (`chat-message.svelte`): consolidate the built-in copy/edit buttons onto a shared `.chat-message-action-button` base rule; give them a visible resting background + border in touch contexts (where hover discoverability is absent); switch the narrow-viewport footer reset from physical `left`/`right` to logical `inset-inline-*`.
2. **Chat input shortcut a11y** (`chat-input.svelte`, `markdown-editor.svelte`): render a visually-hidden shortcut description and point the send button + embedded editor's `aria-describedby` at it; forward native `aria-describedby` through `MarkdownEditor` to both the source `<textarea>` and the WYSIWYG ProseMirror `view.dom` (the focusable surfaces, not the wrapper); set an explicit keycap color so dark-mode contrast passes 4.5:1.
3. **Markdown link popover** (`link-popover.svelte`, `editor-toolbar.svelte`, `markdown-editor.svelte`): anchor the popover with Floating UI (`strategy: 'fixed'`, `bottom-start`, `offset`/`flip`/`shift`) to the toolbar trigger or a ProseMirror-selection virtual element, kept `inert` until the first compute; grow the close button to the 44px touch target; remove the hardcoded `top: 20%` / `left: 50%` / `translateX(-50%)` recipe.
4. **Review comments toggle** (`review-editor-controls.svelte`): render the count via `Badge` (`aria-hidden`), replace `:global(.comments-toggle)` with a locally-scoped selector, and add a polite live announcer for count changes (silent on initial render).
5. **JSON schema editor** (`json-schema-editor-impl.svelte`, `json-schema-toolbar.svelte`): replace the raw bullet diff marker with an `aria-hidden` trailing `Badge` plus sr-only "has changes" text; give the toolbar `role="toolbar"` + an accessible label and roving-tabindex keyboard navigation over enabled actions (including the copy button via DOM discovery).

Also bumps the playground sidebar product gate 85 → 86 to match the current component count (pre-existing overflow shared with `main`; same bump `main` made).

## Review committee

Pre-reviewed by: junior-engineer, simplicity-engineer, Codex (gpt-5.4).
- **Codex** (xhigh round 1 / high rounds 2–3) — 3 rounds; all must-fix items addressed: focus-trap-vs-`inert` interaction (trap now gated on `positionReady`), redundant wrapper `aria-describedby` removed, fragile `void editorState.*` `$effect` reactivity replaced with an explicit `$derived` signature, and live `coordsAtPos` for scroll tracking. APPROVED.
- **junior-engineer** & **simplicity-engineer** — APPROVED after round-1 fixes (roving-index empty-state reset; the `$derived` signature judged a genuine simplification).

> [!NOTE]
> Four committee agents (frontend-architect, ux-designer, testing-expert, typescript-expert) hit a weekly usage limit mid-review and could not complete. Their domains are partially covered by Codex (correctness/types/edge cases) and the TypeScript/lint gates — a human pass on frontend/UX/test architecture is worthwhile.

## Test plan

- Unit + source-contract suites across the five slices — **65 tests pass**.
- `packages/testing/tests/editors-complex-residual.playwright.ts` — **9 browser tests pass** (touch affordance; dark-mode keycap contrast via canvas-resolved oklch; popover anchoring + 44px close button; comments badge/focus; JSE diff indicator + toolbar roving nav).
- `bun run --filter=cinder typecheck` and `lint` — exit 0 (59 pre-existing warnings, 0 errors).
- Full pre-commit suite (`cinder test`, playground) — green.

---

> [!NOTE]
> **Low Risk**
> UI-only accessibility, positioning, and styling changes with broad automated tests; no auth, data, or API surface changes.
> 
> **Overview**
> This PR finishes the **P7 editors & complex** residual audit with five independent accessibility/UX slices, each backed by source-contract, unit, and Playwright coverage.
> 
> **Chat messages** unify copy/edit onto `.chat-message-action-button` with a transparent border so touch/coarse-pointer styles can add a visible surface without layout shift; narrow layouts use `inset-inline-*` for the footer reset.
> 
> **Chat input** adds a screen-reader-only Enter/Shift+Enter description (still available when the visible hint is hidden by container queries), wires `aria-describedby` on the send button and embedded `MarkdownEditor`, and sets explicit keycap `color` for dark-mode contrast.
> 
> **Markdown link popover** anchors with Floating UI (`fixed`, `bottom-start`, flip/shift), keeps the dialog `inert` until positioning completes (focus trap follows), passes the toolbar button or a ProseMirror virtual anchor for Mod-k, and enlarges the close control to the 44px touch token.
> 
> **Review editor** shows comment count via an `aria-hidden` `Badge` (label on the button stays authoritative) and announces count changes with a polite live region; global `.comments-toggle` styling is replaced with a local wrapper.
> 
> **JSON schema editor** replaces the Diff tab raw bullet with sr-only “has changes” plus a trailing `Badge`; the schema toolbar gets `role="toolbar"`, a label, and roving `tabindex` over enabled actions including Copy.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit b8ac3701b4b1bb290e8ca52a8bd0abe1dbf5a70a. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: packages/components/src/components/markdown-editor/markdown-editor.svelte [hunk 1]
```diff
@@ -388,7 +449,9 @@
       onSelectionChange?.(selection);
     },
     onLinkShortcut: () => {
-      // Mod-k pressed - open link popover
+      // Mod-k pressed - open link popover with a virtual element anchor
+      // derived from the current ProseMirror selection position.
+      linkPopoverAnchorElement = resolveLinkPopoverAnchor();
```
[the remark this answers is not in the document]
Comment (stevekinney) [reply packages/components/src/components/markdown-editor/markdown-editor.svelte:465]: Fixed in commit 6a1f9296: onLinkShortcut now captures capturedLinkRange = lastKnownLinkRange before opening the popover, matching handleLinkClick.

File: packages/components/src/components/markdown-editor/markdown-editor.svelte [hunk 2]
```diff
@@ -302,14 +306,69 @@
...
+                y: coords.top,
+                width: coords.right - coords.left,
+                height: coords.bottom - coords.top,
+                top: coords.top,
+                right: coords.right,
+                bottom: coords.bottom,
+                left: coords.left,
+              };
+            },
+          };
```
[the remark this answers is not in the document]
Comment (stevekinney) [reply packages/components/src/components/markdown-editor/markdown-editor.svelte:357]: Fixed in commit 6a1f9296: getBoundingClientRect callback now wraps view.coordsAtPos() in a try/catch, falling back to view.dom.getBoundingClientRect() so autoUpdate cycles can't throw and leave the popover stuck.
